torch/csrc/distributed/c10d/TCPStore.cpp

```diff
@@ -1148,16 +1170,34 @@ void TCPStore::wait(
 void TCPStore::doWait(
     c10::ArrayRef<std::string> keys,
     std::chrono::milliseconds timeout) {
-  // TODO: Should we revert to the original timeout at the end of the call?
-  client_->setTimeout(timeout);
-
   client_->sendCommand(detail::QueryType::WAIT);
   client_->sendStrings(keys);
 
-  auto response = client_->receiveValue<detail::WaitResponseType>();
-  if (response != detail::WaitResponseType::STOP_WAITING) {
-    TORCH_CHECK(false, "Stop_waiting response is expected");
+  detail::WaitResponseType response;
+  if (client_->receiveValueWithTimeout<detail::WaitResponseType>(
+          response, timeout)) {
+    if (response != detail::WaitResponseType::STOP_WAITING) {
+      TORCH_CHECK(false, "Stop_waiting response is expected");
```

Comment:
will we hard stop here (if we receive STOP_WAIT we'll return, if we receive WAIT_CANCELED we'll fail the torch check, so whatever below this is not run?

Reply:
Yes, exactly.

It would be a bug to receive WAIT_CANCELED since at this point the client has not sent a CANCEL_WAIT command yet.